pkg/virt-controller/watch/vmi.go

```diff
@@ -1056,6 +1095,18 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 	}
 
 	if !isTempPod(pod) && isPodReady(pod) {
+		if vmispec.SRIOVInterfaceExist(vmi.Spec.Domain.Devices.Interfaces) {
+			networkPCIMapAnnotationValue := sriov.CreateNetworkPCIAnnotationValue(
+				vmi.Spec.Networks, vmi.Spec.Domain.Devices.Interfaces, pod.Annotations[networkv1.NetworkStatusAnnot],
+			)
+			newAnnotations := map[string]string{sriov.NetworkPCIMapAnnot: networkPCIMapAnnotationValue}
+			patchedPod, err := c.syncPodAnnotations(pod, newAnnotations)
+			if err != nil {
+				return &syncErrorImpl{err, FailedPodPatchReason}
+			}
+			*pod = *patchedPod
```

Comment:
I think that changing it as follows will make the code easier to understand, WDYT?
```
var err error
...
pod, err = c.syncPodAnnotations(pod, newAnnotations)
...
Or
if pod, err := c.syncPodAnnotations(pod, newAnnotations); err != nil {
...
```

Reply:
I don't think so.
the whole point is for `pod` to adjust to the changes made but NOT change its pointer address. 
patchedPod is a temporary variable that will loose context outside of this scope.

Reply:
> I think that changing it as follows will make the code easier to understand, WDYT?
> 
> ```
> var err error
> ...
> pod, err = c.syncPodAnnotations(pod, newAnnotations)
> ...
> ```

I think it's better to explicitly say that we are changing the pod to be patched.

Reply:
> I think that changing it as follows will make the code easier to understand, WDYT?
> 
> ```
> var err error
> ...
> pod, err = c.syncPodAnnotations(pod, newAnnotations)
> ...
> ```

I think this way is more explicit to suggest that the pod object is changed. but I'm kinda on the fence since it's not common practice..